pkg/scheduler/server/internal/etcd/etcd.go

```diff
@@ -135,6 +142,38 @@ func (e *etcd) Client(ctx context.Context) (*clientv3.Client, error) {
 	}
 }
 
+func (e *etcd) runDefrag(ctx context.Context) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-time.After(10 * time.Minute):
+			log.Debug("Checking if Etcd needs Defragmentation")
+			resp, err := e.client.Maintenance.Status(ctx, e.config.ListenClientUrls[0].Host)
+			if err != nil {
+				return err
```

Comment:
Will returning an error here and anywhere else in this method not cause the `runDefrag` logic to stop forever and ot repeat anymore when the cause of the error might be transient?

Reply:
Yeah, just trying to prevent a situation where transient errors cause a destructive event like scheduler shutdown

Reply:
:+1: done!